src/components/shared/TractionPagination.vue

```diff
@@ -0,0 +1,194 @@
+          data-testid="page-button"
+          @click="pageClick(page.name)"
+        >
+          {{ page.name }}
+        </traction-button>
+      </div>
+
+      <TractionButton
+        theme="paginationDefault"
+        :disabled="isInLastPage"
+        data-testid="next-button"
+        @click="nextPageClick"
+      >
+        <traction-pagination-icon>
+          <path d="M9.4 18 8 16.6l4.6-4.6L8 7.4 9.4 6l6 6Z" />
+        </traction-pagination-icon>
+      </TractionButton>
+      <TractionButton
+        theme="paginationDefault"
+        :disabled="isInLastPage"
+        data-testid="last-button"
+        @click="lastPageClick"
+      >
+        <traction-pagination-icon>
+          <path
+            d="M6.4 18 5 16.6 9.575 12 5 7.4 6.4 6l6 6Zm6.6 0-1.4-1.4 4.575-4.6L11.6 7.4 13 6l6 6Z"
+          />
+        </traction-pagination-icon>
+      </TractionButton>
+    </flagged-feature>
+  </div>
+</template>
+
+<script>
+import { BPagination } from 'bootstrap-vue'
+
+export default {
+  /**
+   * # TractionPagination
+   *
+   * Displays a custom pagination or bootstrap component based on 'enable_custom_pagination' feature flag
+   *
+   */
+  name: 'TractionPagination',
+  components: { BPagination },
+  inheritAttrs: false,
+  props: {
+    //value field  which will be bind automatically with 'v-model' prop passed into the component
+    value: {
+      type: Number,
+      default: 1,
+    },
+
+    /**How many total items are in the list */
+    totalRows: {
+      type: Number,
+      default: 0,
+    },
+    /**Number of items that every page represents */
+    perPage: {
+      type: Number,
+      default: 10,
+    },
+    /**Maximum visible page buttons to be displayed */
+    maxVisibleButtons: {
+      type: Number,
+      default: 3,
+    },
+  },
+  data() {
+    return {
+      currentPage: this.value,
+    }
+  },
+  computed: {
+    //Calculate total pages required
+    totalPages() {
+      return Math.ceil(this.totalRows / this.perPage)
+    },
+    //If total number of pages is less than number of buttons given, display only as many buttons as the pages
+    visibleButtons() {
+      return Math.min(this.totalPages, this.maxVisibleButtons)
+    },
+    //Calculate the start page number to be displayed (in page button), based on current selectiobn
+    startPage() {
+      if (this.currentPage === 1) {
+        return 1
+      }
+
+      if (this.currentPage === this.totalPages) {
+        return this.totalPages - this.visibleButtons + 1
+      }
+
+      return this.currentPage - 1
+    },
+    //Calculate the end page number to be displayed, based on current selection
+    endPage() {
+      return Math.min(this.startPage + this.visibleButtons - 1, this.totalPages)
+    },
+    //Get page numbers to display in page-buttons
+    pages() {
+      const range = []
+      for (let i = this.startPage; i <= this.endPage; i++) {
+        range.push({
+          name: i,
+        })
+      }
+      return range
+    },
```

Comment:
there is a nifty way where you can use Array.from. In this case it would be 
`Array.from({length: this.endPage - this.startPage}, (v, k) => k+this.startPage)`
Not necessary but cleaner maybe ...

Reply:
Sure, that's a good suggestion. I will do it

Reply:
@stevieing I have updated the review comments.